kernel/kernel_drivers.go

```diff
@@ -422,9 +423,21 @@ func NeedsKernelDriversTree(mod *asserts.Model) bool {
 	// TODO this won't work for a UC2{0,2} -> UC24+ remodel as we need the
 	// new model here. Get to this ASAP after snapd 2.62 release.
 	switch mod.Base() {
-	case "core20", "core22", "core22-desktop":
+	case "core22":
+		if mod.Classic() {
+			// This is a workaround for LP#2104933. The base should
+			// never have been core22 in 24.04/24.10.
+			return classicModelWithWrongBase()
+		}
+		return false
+	case "core20", "core22-desktop":
 		return false
 	default:
 		return true
 	}
 }
+
+func classicModelWithWrongBase() bool {
```

Comment:
I know there is a comment above but probably we want one here too, and call this classicModel24XXWrongBase perhaps?

Reply:
Added now